src/connections/Validation.ts

```diff
@@ -0,0 +1,142 @@
+import { isEmpty } from "lodash";
+import { useCallback, useState } from "react";
+
+import { v3Resource } from "@/connections/util/apiConnectionFactory";
+import {
+  getPolygonValidation,
+  GetPolygonValidationPathParams,
+  getSiteValidation
+} from "@/generated/v3/researchService/researchServiceComponents";
+import { ValidationDto } from "@/generated/v3/researchService/researchServiceSchemas";
+import { useConnection } from "@/hooks/useConnection";
+import ApiSlice from "@/store/apiSlice";
+import { loadConnection } from "@/utils/loadConnection";
+
+const hasValidParams = (pathParams: GetPolygonValidationPathParams | undefined): boolean => {
+  const isValid =
+    pathParams != null &&
+    pathParams.polygonUuid != null &&
+    pathParams.polygonUuid !== "" &&
+    pathParams.polygonUuid !== "undefined" &&
+    !isEmpty(pathParams.polygonUuid);
+
+  return isValid;
+};
+
+const validationConnection = v3Resource("validations", getPolygonValidation)
+  .singleResource<ValidationDto>(({ id }) => {
+    if (id == null) return undefined;
+    return { pathParams: { polygonUuid: id } };
+  })
+  .enabledProp()
+  .buildConnection();
+
+export const usePolygonValidation = (pathParams: GetPolygonValidationPathParams) => {
+  const result = useConnection(validationConnection, {
+    id: pathParams.polygonUuid,
+    enabled: hasValidParams(pathParams)
+  });
+
+  const validationData = result[1].data;
+
+  return validationData;
+};
+
+const siteValidationConnection = v3Resource("validations", getSiteValidation)
+  .index<ValidationDto>(() => ({
+    pathParams: { siteUuid: "" },
+    queryParams: { page: {} }
+  }))
+  .pagination()
+  .enabledProp()
+  .addProps<{ siteUuid: string; criteriaId?: number }>(({ siteUuid, criteriaId }) => ({
+    pathParams: { siteUuid },
+    queryParams: { page: {}, ...(criteriaId != null && { criteriaId }) }
+  }))
```

Comment:
This is OK, but it's a little more in keeping with other factory uses to define this in the call to `.index`:
```
.index<ValidationDto, { siteUuid: string, criteriaId?: number }>(({ siteUuid, criteriaId }) => ({ 
  pathParams: { siteUuid },
  queryParams: { page: {}, criteriaId }
})
```

Reply:
It's a little weird to always pass an empty object for page. Is that correct? I think `pagination()` factory method should be handling that for you, so you can likely just leave it off here.